Sources/Apodini/Properties/Parameter.swift

```diff
@@ -50,16 +55,17 @@ public struct Parameter<Element: Codable>: Property {
     ///   - options: Options passed on to different interface exporters to clarify the functionality of this `@Parameter` for different API types
     public init(_ name: String, _ options: Option...) {
         self.defaultValue = nil
-        self.name = name
         self.options = PropertyOptionSet(options)
+        defer { // swiftlint:disable:this inert_defer
+            self.name = name
+        }
```

Comment:
I am not sure why you use a defer statement here. As this would be the last operation it will be executed last anyways the defer would not be needed here.
```suggestion
        self.name = name
```

Reply:
The defer is intentional, as otherwise the `didSet` will not execute. `didSet` or `willSet` are not executed for values set within the initializer. So my decision was to use defer instead of creating some method which is called in every initializer. This would also work for me if defer isn't considered a good solution. Or we move the empty name check into the `EndpointParameter` or something.

Reply:
Oh, this is an interesting behavior. If we decide to keep this I would recommend to add comment explaining the behavior. 
But I might have a better solution for the problem.
I would suggest to add a private initializer that includes the precondition and make all properties except of `element` constants. All other initializers then call the private initializer. The 
```swift
    let id: UUID
    let name: String?
    let options: PropertyOptionSet<ParameterOptionNameSpace>
    let defaultValue: Element?
    
    private var element: Element?

    // ...

    public init() {
            // The only time where we need to pass in a parameter even though it is not an input of the initializer, otherwise we would have an endless loop.
        self.init(id: UUID())
    }

    // All other initializer call the private initializer too.
    private init(id: UUID = UUID(), defaultValue: Element? = nil, name: String? = nil, options: PropertyOptionSet<ParameterOptionNameSpace> = PropertyOptionSet()) {
        if let name = name {
            precondition(!name.isEmpty, "The name for Parameter cannot be empty!")
        }
        
        self.id = id
        self.defaultValue = defaultValue
        self.name = name
        self.options = options
    }
```
The private initializer then includes the precondition check and as all properties are constants we don't need to care about any possible updates to the `name` property.

Reply:
yeah of course. I could have tried to think of simpler solutions like yours 😄
I guess I just wanted to try out `didSet` and make it work 🙃

I think the unified initializer is the best solution 👍 Thanks